22.cpp

```diff
@@ -0,0 +1,41 @@
+#include <iostream>
+#include <stack>
+#include <string>
+#include <vector>
+
+class Solution {
+ public:
+  std::vector<std::string> generateParenthesis(int n) {
+    std::stack<char> matching_sta;
```

Comment:
Note `matching_sta` only stores `(`, so a integer (rather than a stack) is enough!

Reply:
changed